pkg/proxy/ipvs/proxier.go

```diff
@@ -919,6 +945,12 @@ func (proxier *Proxier) syncProxyRules() {
 		klog.V(4).Infof("syncProxyRules took %v", time.Since(start))
 	}()
 
+	localAddrs, err := getLocalAddrs()
+	if err != nil {
+		klog.Errorf("Failed to get local addresses during proxy sync: %v", err)
+		return
```

Comment:
One change in behavior worth mentioning. Today, if we fail to get the local addresses, we just assume the external IP isn't part of local addresses. Now we log error and return early. Wondering if we need to preserve that behavior.

Reply:
Maybe don't return here, and on line 1135, do a check for `len(localAddrs)`. If `len(localAddrs) == 0`, assume externalIP is not local.

Reply:
thanks for catching this, we should continue syncing when get local adds failed for case that there is no external ip service.